### PR TITLE
Added interface to re-export module with theme interface

### PR DESCRIPTION
**Pull request by davidruisinger**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present) (I was getting an "Cannot find module 'csstype'" error from index.d.ts which I didn't touch)

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.styled-components.com/docs/api#typescript
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/styled-components/index.d.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@
 //                 Adam Lavin <https://github.com/lavoaster>
 //                 Jessica Franco <https://github.com/Jessidhia>
 //                 Jason Killian <https://github.com/jkillian>
+//                 David Ruisinger <https:/>/github.com/flavordaaave>
```
Comment (JKillian) [inline types/styled-components/index.d.ts]: Looks like a typo in this line

**Review by jkillian — CHANGES_REQUESTED** [review]
Could you add a test, similar to the tests we have in the non-native styled-components test file?

**Comment by flavordaaave** [comment]
@JKillian I've added some tests (I more or less copied the tests over from `index.tsx` and adjusted them where necessary).

**Comment by flavordaaave** [comment]
@JKillian I resolved the merge conflicts. Could you please have another look at it? Thanks!

File: types/styled-components/test/native.tsx [hunk 2]
```diff
@@ -57,3 +58,141 @@ const TomatoButton = styled(MyButton)`
...
+  const ThemedView3 = styled.View(props => {
+    // $ExpectType string
+    props.theme.color;
+    // $ExpectType string | undefined
+    props.testID;
+
+    return css`
+        background: ${props.theme.color};
+    `;
+});
```
Comment (JKillian) [inline types/styled-components/test/native.tsx]: indentation got a bit messed up starting with this line

**Review by jkillian — APPROVED** [review]
Thanks for adding tests, looks good to me! Did notice some indentation got a bit messed up, so if you have time it'd be nice to fix that

**Pull request merged**